core/include/detray/masks/cylinder3.hpp

```diff
@@ -80,21 +80,17 @@ class cylinder3 final
     }
 
     /** Mask operation
-     *
-     * @tparam inside_local_type::point3 is the deduced type of the point to be
-     *checked w.r.t. to the mask bounds, it's assumed to be within the cylinder
-     *3D frame
      *
      * @param p the point to be checked
      * @param t is the tolerance tuple in (radius, z)
      *
      * @return an intersection status e_inside / e_outside
      **/
-    template <typename inside_local_t, bool is_rad_check = kRadialCheck>
+    template <typename cartesian_point_t>
     DETRAY_HOST_DEVICE intersection::status is_inside(
-        const point3 &p,
+        const cartesian_point_t &p,
         const scalar t = std::numeric_limits<scalar>::epsilon()) const {
-        if constexpr (is_rad_check) {
+        if constexpr (kRadialCheck) {
```

Comment:
This was needed in the helix unittest unfortunately. I can check again, whether the newton algorithm correctly converges, if this is set to ```false``` (which it is in the toy geometry), but from memory, it didn't. We should rather leave the template on the function call, instead of the mask